src/include/misc.i

```diff
@@ -248,3 +248,26 @@ __wt_spin_backoff(uint64_t *yield_count, uint64_t *sleep_usecs)
 	(*sleep_usecs) = WT_MIN((*sleep_usecs) + 100, WT_THOUSAND);
 	__wt_sleep(0, (*sleep_usecs));
 }
+
+				/* Maximum stress delay is 1/10 of a second. */
+#define	WT_TIMING_STRESS_MAX_DELAY	(100000)
+
+/*
+ * __wt_timing_stress --
+ *	Optionally add delay to stress code paths.
+ */
+static inline void
+__wt_timing_stress(WT_SESSION_IMPL *session, u_int flag)
+{
+	WT_CONNECTION_IMPL *conn;
+	uint64_t sleep_usecs;
+
+	conn = S2C(session);
+
+	/* Only sleep when the specified configuration flag is set. */
+	if (!FLD_ISSET(conn->timing_stress_flags, flag))
+		return;
+
+	sleep_usecs = __wt_random(&session->rnd) % WT_TIMING_STRESS_MAX_DELAY;
```

Comment:
I like this idea - I actually think it'd be very cool if it could skew low so the pause is generally very short. A concern I have with this mode is that it slows things down so much we get less real work done and thus split races aren't as common..

Reply:
I did a uniform distribution because it was easy. You might be correct, but it's hard to know for sure. Generally in my own testing I use big sleeps, but I have no idea if that's optimal or not...